src/biz/metrics.go

```diff
@@ -276,19 +249,62 @@ func (mc *MetricsFetcher) memory(mem raw.Memory) Memory {
 		softLimit = 0
 	}
 
-	return Memory{
-		MemLimitBytes:         memLimits,
-		CacheUsageBytes:       mem.Cache,
-		RSSUsageBytes:         mem.RSS,
-		UsageBytes:            mem.RSS,
-		UsagePercent:          usagePercent,
-		KernelUsageBytes:      mem.KernelMemoryUsage,
-		SwapUsageBytes:        swapUsage,
-		SwapOnlyUsageBytes:    swapOnlyUsage,
-		SwapLimitBytes:        swapLimit,
-		SoftLimitBytes:        softLimit,
-		SwapLimitUsagePercent: swapLimitUsagePercent,
+	swapLimit := mem.SwapLimit
+	if mem.SwapLimit > math.MaxInt64/2 {
+		swapLimit = 0
+	}
+
+	m := Memory{
+		MemLimitBytes:    memLimits,
+		CacheUsageBytes:  mem.Cache,
+		RSSUsageBytes:    mem.RSS,
+		UsageBytes:       mem.RSS,
+		UsagePercent:     usagePercent,
+		KernelUsageBytes: mem.KernelMemoryUsage,
+		SoftLimitBytes:   softLimit,
+		SwapLimitBytes:   swapLimit,
+	}
+
+	if mem.SwapUsage != nil {
+		m = populateSwapUsageMetrics(mem, m, swapLimit)
+	}
+
+	return m
+}
+
+/*
+	https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
+
+For efficiency, as other kernel components, memory cgroup uses some optimization
+to avoid unnecessary cacheline false sharing. usage_in_bytes is affected by the
+method and doesn't show 'exact' value of memory (and swap) usage, it's a fuzz
+value for efficient access. (Of course, when necessary, it's synchronized.)
+If you want to know more exact memory usage, you should use RSS+CACHE(+SWAP)
+value in memory.stat(see 5.2).
+However, as the `docker stats` cli tool does, page cache is intentionally
+excluded to avoid misinterpretation of the output.
+Also mem.SwapUsage is parsed from memory.memsw.usage_in_bytes, which
+according to the documentation reports the sum of current memory usage
+plus swap space used by processes in the cgroup (in bytes). That's why
+Usage is subtracted from the Swap: to get the actual swap.
+*/
+func populateSwapUsageMetrics(mem raw.Memory, m Memory, swapLimit uint64) Memory {
+	var swapOnlyUsage uint64
+	if *mem.SwapUsage != 0 { // for systems that have swap disabled
```

Comment:
shouldnt check nil pointer?

Reply:
I think it's checked outside the function. However, maybe I'd like this to be self-contained in the function itself to avoid potential nil panic if we use that function somewhere else and we forget to check for `nil`.

Reply:
yes that function is called just in one place if mem.SwapUsage != nil. 

I can add the guard also inside, but it felt like a bit too much